clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerNetworkClient.java

```diff
@@ -404,6 +411,12 @@ private void maybeTriggerWakeup() {
             throw new WakeupException();
         }
     }
+    
+    private void maybeThrowInterruptException() {
```

Comment:
About catching `WakeupException`: Thanks, that makes sense.

About `wakeup`/`close` races: Okay, that makes sense, though I think it should usually still be possible to put the "shut down consumer thread" flag check into the `WakeupException` catch block, and have the polling loop be of the `while (true)` form (there are probably some cases I'm not thinking of). The KafkaConsumer javadoc seems pretty clear that `wakeup` is the preferred method for interrupting a consumer, but let me know if you'd like a note added somewhere. 

Regarding `close`, you're right that it's only a problem if auto-commit is enabled, but that's still not ideal, since you can then safely interrupt a consumer thread except if it's in auto-commit mode, and happens to be closing and there are offsets to commit, in which case it'll hang.